specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/preview/2022-07-01-preview/healthcare-apis.json

```diff
@@ -4093,6 +4138,224 @@
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "MedtechMappingValidationCategory": {
+      "type": "string",
+      "description": "Indicates the type of validation being performed",
+      "enum": [
+        "NORMALIZATION",
+        "FHIRTRANSFORMATION"
+      ],
+      "x-ms-enum": {
+        "name": "MedtechMappingValidationCategory",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "MedtechMappingValidationErrorLevel": {
+      "type": "string",
+      "description": "Indicates the severity of the validation error",
+      "enum": [
+        "ERROR",
+        "WARN"
+      ],
+      "x-ms-enum": {
+        "name": "MedtechMappingValidationErrorLevel",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "MedtechMappingValidationError": {
+      "type": "object",
+      "description": "A mapping validation error.",
+      "readOnly": true,
+      "properties": {
+        "message": {
+          "description": "The validation error message.",
+          "type": "string"
+        },
+        "level": {
+          "$ref": "#/definitions/MedtechMappingValidationErrorLevel"
+        },
+        "category": {
+          "$ref": "#/definitions/MedtechMappingValidationCategory"
+        },
+        "lineInfo": {
+          "$ref": "#/definitions/MedtechMappingValidationLineInfo"
+        }
+      }
+    },
+    "MedtechMappingValidationErrorList": {
+      "type": "array",
+      "description": "A list of mapping validation error.",
+      "x-ms-identifiers": [],
+      "readOnly": true,
+      "items": {
+        "$ref": "#/definitions/MedtechMappingValidationError"
+      }
+    },
+    "ValidateMedtechMappingsDeviceResult": {
+      "description": "A result returned when processing a single device event in Medtech Mapping Validation Service.",
+      "type": "object",
+      "properties": {
+        "aggregatedCount": {
+          "description": "Indicates how many Device Events produced the associated Exception.",
+          "type": "object"
+        },
+        "deviceEvent": {
+          "description": "The device event which was validated.",
+          "type": "object"
+        },
+        "exceptions": {
+          "$ref": "#/definitions/MedtechMappingValidationErrorList"
+        },
+        "measurements": {
+          "description": "A collection of measurements which were produced.",
+          "type": "array",
+          "x-ms-identifiers": [],
+          "items": {
+            "description": "A measurement.",
+            "$ref": "#/definitions/MedtechMeasurement"
+          }
+        },
+        "observations": {
+          "description": "A collection of observations which were produced.",
+          "type": "array",
+          "x-ms-identifiers": [],
+          "items": {
+            "description": "An observation.",
+            "type": "object"
```

Comment:
same here please define the schema

Reply:
The returned object will be a Fhir Observation. This has a well known schema which is defined [here](https://www.hl7.org/fhir/observation.html#resource). Duplicating this schema here and keeping it up to date is possible but would be a very large undertaking. Is it enough to point to this objects schema definition inside of the `description`?

Reply:
Having a schema for the resources is a hard requirement for the apis in the control plane. 

there are some issues with how this is modelled : 
- Having free form schemas mean that clients do not have a consistent model to code against , each time you return a different schema they may not know how to handle it. 
- This is made worse during versioning since there is no way to version these either so clients do not even have a baseline against which to operate, you cannot even ask them to use an older client since there is nothing that enforces the schema across versions either. 
- There is no validation for client side possible with these data 
- There is a hard limit on the size of this data that ARM can handle. Freeform data tends to grow larger so you might end up hitting those limits

From your description above and the other comment , have you really considered if this data really belongs in the control plane ? 
You could consider exposing this data to the user in some other way ? Maybe uploading to user provided data store since this seems to be an issue for some report value?

Reply:
@raosuhas 

We have already reviewed and received approval for this API in private preview and released it. This is a follow up to add for public preview.  That PR is available here https://github.com/Azure/azure-rest-api-specs-pr/pull/7446.

JSON object is the correct and appropriate schema for this API.  The content is the same as the existing GA properties found [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/stable/2022-06-01/healthcare-apis.json#L3556) and [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/stable/2022-06-01/healthcare-apis.json#L3635).  We received the necessary VP approval for the exemption.  I can discuss the specifics if necessary.

If the concern is this being a public API I am open to discuss alternatives, similar to what exists for other resources like Event Grid. While we see benefits in allowing the validation API to be called by end users it isn't the core use case.

Reply:
Looks like the issue seemed to have been missed in the previous PR and was not discussed there. The VP exception is not really general accepted practice for these cases. I have started an email thread with the relevant folks. and added you both on it. We can follow up there